### DISCUSSION TITLE
Will this code trigger provider notifications?

### DISCUSSION THREAD

**Discussion by busslina** [rrousselGit/riverpod#3083] [Q&A]
If `newState` is equals to `state.requireValue` does this instruction triggers provider notifications?
In other words, will it be considered as a new state?

```dart

// Updating state
state = AsyncData(newState);
```

`AsyncValue`'s `==` operator. Seems to not include value comparison:
![imagen](https://github.com/rrousselGit/riverpod/assets/73592852/55d53f20-36bd-41b4-9639-f3f7621cb2a2)

**Comment by rrousselGit** [comment]
> AsyncValue's == operator. Seems to not include value comparison:

Value comparison is done with `valueOrNull`. Is that what you're referring to?

**Reply by busslina** [reply]
It was that, thanks
